src/test/java/racingcar/domain/RaceParticipantsTest.java

```diff
@@ -0,0 +1,77 @@
+package racingcar.domain;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import racingcar.domain.car.Car;
+import racingcar.domain.car.move.MovingStrategy;
+import racingcar.dto.request.RaceParticipantsRequest;
+import racingcar.exception.InvalidInputException;
+import racingcar.mock.MockMovingStrategy;
+
+class RaceParticipantsTest {
+    private MovingStrategy mockMovingStrategy;
+
+    @BeforeEach
+    void setUp() {
+        mockMovingStrategy = new MockMovingStrategy();
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"a,b,b", "a,b,c,b", "a,b,c,b,c"})
+    void 중복된_자동차_이름이_있을_경우_실패(String name) {
+        //when
+        final RaceParticipantsRequest raceParticipantsRequest = new RaceParticipantsRequest(name);
+
+        //then
+        assertThatThrownBy(() -> raceParticipantsRequest.toRaceParticipants(mockMovingStrategy))
+                .isInstanceOf(InvalidInputException.class);
+    }
+
+    @Nested
+    class 우승자를_가려낸다 {
+        @Test
+        void 우승자가_한_명인_경우() {
+            //given
+            final Car car1 = new Car("car1", new MockMovingStrategy(List.of(true, true, true)));
+            final Car car2 = new Car("car2", new MockMovingStrategy(List.of(true, true, false)));
+            final Car car3 = new Car("car3", new MockMovingStrategy(List.of(true, false, false)));
+
+            final RaceParticipants raceParticipants = new RaceParticipants(List.of(car1, car2, car3));
+            for (int i = 0; i < 3; i++) {
+                raceParticipants.move();
+            }
+
+            //when
+            final List<Car> raceWinners = raceParticipants.getRaceWinners();
+
+            //then
+            assertThat(raceWinners).hasSize(1).isEqualTo(List.of(car1));
+        }
+
+        @Test
+        void 우승자가_여러_명인_경우() {
+            //given
+            Car car1 = new Car("car1", new MockMovingStrategy(List.of(true, true, true)));
+            Car car2 = new Car("car2", new MockMovingStrategy(List.of(true, true, true)));
+            Car car3 = new Car("car3", new MockMovingStrategy(List.of(true, false, false)));
```

Comment:
🖐🏻 조금 더 의미있는 변수명을 지으면 어떨까요?

`테스트는 문서의 역할`을 하기도 하는데요! **car1,2,3..** 를 보고서는 우승한 자동차인지, 패배한 자동차인지 구분하기 어렵고, 그러면 뒤에 코드까지 다 읽어야만 파악할 수 있는 코드가 되는 것 같아요.

Reply:
```suggestion
            Car winningCar = new Car("car1", new MockMovingStrategy(List.of(true, true, true)));
            Car losingCar1 = new Car("car2", new MockMovingStrategy(List.of(true, true, true)));
            Car losingCar2 = new Car("car3", new MockMovingStrategy(List.of(true, false, false)));
```
로 하면 좋겠군요.
핑계를 대자면 new MockMovingStrategy 부분의 위치를 맞출려고 car1, car2, car3로 했습니다...